velox/exec/RowContainer.h

```diff
@@ -384,7 +387,25 @@ class RowContainer {
       int32_t numRows,
       int32_t columnIndex,
       const VectorPtr& result) {
-    extractColumn(rows, numRows, columnAt(columnIndex), result);
+    extractColumn(
+        rows,
+        numRows,
+        columnAt(columnIndex),
+        columnHasNulls(columnIndex),
+        result);
+  }
+
+  /// Copies the values at 'columnIndex' into 'result' for the 'numRows' rows
+  /// pointed to by 'rows'. If an entry in 'rows' is null, sets corresponding
+  /// row in 'result' to null. If 'columnHasNulls' is false, null free
+  /// optimization would be applied.
+  void extractColumn(
+      const char* const* rows,
+      int32_t numRows,
+      int32_t columnIndex,
+      bool columnHasNulls,
```

Comment:
I'm thinking that it makes sense to pass columnHasNulls flag only to static methods, because, when we fetch data from "this" RowContainer then we should let the container to decide whether column has nulls or not.

It would be nice to add comments to static methods that take columnHasNulls flag to explain why they take it and that it is the callers responsibility to ensure the flag is correct.